### ISSUE TITLE
Optimizable properties for terminal nodes in a tensor expression

### ISSUE THREAD

**Issue by campsd** [yhtang/FunFact#38]
Currently all `tensors` in a tensor expression are updated in the gradient descent step of a factorization model. We should implement a feature such that the user can decide which of the `tensors` (and maybe also `LiteralValues`?)  are actually optimized.

Potential syntax:
```
A = ff.tensor('A', 2, 3, optimizable=True)
offset = ff.literal('o', optimizable=True)
...
```

**Comment by yhtang** [comment]
In addition to explicitly specifying the `optimizable` attribute, we can also assume by default that an abstract tensor instantiated with only shape information will be optimizable (since they will be random initialized), while one made from a concrete tensor will be constant, e.g.
```
A = ff.tensor('A', 2, 3, 4)  # A defaults to optimizable
B = ff.tensor('B', np.eye(3))  # B defaults to non-optimizable.
```

**Comment by yhtang** [comment]
Also makes sense to provide `.set_optimizable(True/False)` methods for the terminal nodes.

**Comment by campsd** [comment]
This is works nicely with `torch` where optimizable tensors have to be created by setting `requires_grad=True`

**Comment by yhtang** [comment]
> This is works nicely with `torch` where optimizable tensors have to be created by setting `requires_grad=True`

Yep. What is the JAX equivalence of it?

**Comment by campsd** [comment]
I think not returning the non-optimizable leafs in `Factorize.factors` so JAX can't trace them? I'm working on implementing this feature now.

**Comment by yhtang** [comment]
> I think not returning the non-optimizable leafs in `Factorization.factors` so JAX can't trace them? I'm working on implementing this feature now.

I see. Makes sense. Thanks!

**Comment by yhtang** [comment]
Implemented by #97.
